serve/graph/schema/filter/transaction_filter.graphql

```diff
@@ -0,0 +1,240 @@
+"""
+Filters for querying Transactions within specified criteria related to their execution and placement within Blocks.
+"""
+input TransactionFilter {
+  """
+  Minimum block height from which to start fetching Transactions, inclusive. Aids in scoping the search to recent Transactions.
+  """
+  from_block_height: Int
+
+  """
+  Maximum block height up to which Transactions should be fetched, exclusive. Helps in limiting the search to older Transactions.
+  """
+  to_block_height: Int
+
+  """
+  Minimum Transaction index from which to start fetching, inclusive. Facilitates ordering in Transaction queries.
+  """
+  from_index: Int
+
+  """
+  Maximum Transaction index up to which to fetch, exclusive. Ensures a limit on the ordering range for Transaction queries.
+  """
+  to_index: Int
+
+  """
+  Minimum `gas_wanted` value to filter Transactions by, inclusive. Filters Transactions based on the minimum computational effort declared.
+  """
+  from_gas_wanted: Int
+
+  """
+  Maximum `gas_wanted` value for filtering Transactions, exclusive. Limits Transactions based on the declared computational effort.
+  """
+  to_gas_wanted: Int
+
+  """
+  Minimum `gas_used` value to filter Transactions by, inclusive. Selects Transactions based on the minimum computational effort actually used.
+  """
+  from_gas_used: Int
+
+  """
+  Maximum `gas_used` value for filtering Transactions, exclusive. Refines selection based on the computational effort actually consumed.
+  """
+  to_gas_used: Int
+
+  """
+  Hash from Transaction content in base64 encoding. If this filter is used, any other filter will be ignored.
+  """
+  hash: String
+
+  """
+  Transaction's message to filter Transactions.
+  """
+  message: TransactionMessageInput
+
+  """
+  `memo` value to filter Transaction's memo.
+  """
+  memo: String
+}
+
+"""
+Input for filters by transaction message.
+"""
+input TransactionMessageInput {
+  """
+  The type of transaction message.
+  """
+  type_url: MessageType
+
+  """
+  The route of transaction message.
+  """
+  route: MessageRoute
+
+  """
+  `TransactionBankMessageInput` represents input parameters required when the message router is `bank`.
+  """
+  bank_param: TransactionBankMessageInput
+
+  """
+  `TransactionVmMessageInput` represents input parameters required when the message router is `vm`.
+  """
+  vm_param: TransactionVmMessageInput
```

Comment:
I think it's a good idea. 

I thought about making it simpler. 
But it seemed like it would be hard to guarantee the structure of the transaction message.

For example.
Currently, only the `bank` router has messages of type `send`, but it is possible that the `vm` router will also have messages of type `send`. In this case, it would be the same `send` type, but with a different payload.

Of course, this is very unlikely to happen, but I thought we should also consider exceptions.

Reply:
We can even simplify more:

```go
query {
    transactions(filter:{
        // route: vm // you can still filter by vm or bank
        params: {
              from_address: "VAL"
              caller: "CALLER"
        }
    })
}
```

Now, we don't need to care about having different send parameters with different payloads. With this approach, every transaction with a param `from_address!=VAL AND caller!=CALLER` will be filtered, even if we have these same params on different kinds of messages.

Reply:
It certainly seems simple enough.
But I'm worried about confusing the filter parameters.

My concern is twofold 
1. users should be able to recognize exactly what parameters they are filtering. 
For example, amount in BankMsgSend, send in MsgCall, etc., can be missed if they are separated by variable names, and they need to know the description and structure.

2. It is difficult to distinguish conditions between parameters with the same name. 
For example, if you only want to filter messages from MsgCall and MsgPackage by package path, there is no way to filter messages from MsgRun.

Can you share your thoughts ?

Reply:
I see your point. With my approach won't be possible to get specific message types that are not sharing params. LGTM.